exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -0,0 +1,245 @@
+ * (with no real network setup). Although it is possible to add several validator nodes to this
+ * network, only one node will create the service instances and will execute their operations
+ * (e.g., {@link Service#afterCommit(BlockCommittedEvent)} method logic).
+ *
+ * <p>When TestKit is created, Exonum blockchain instance with given services is initialized and
+ * genesis block is committed. The service instances are created and their
+ * {@linkplain UserServiceAdapter#initialize(long)}  initialization} methods are called and
+ * {@linkplain UserServiceAdapter#mountPublicApiHandler(long)} public API handlers} are created.
+ *
+ * @see <a href="https://exonum.com/doc/version/0.10/get-started/test-service/">TestKit documentation</a>
+ */
+public final class TestKit {
+
+  @VisibleForTesting
+  static final short MAX_SERVICE_NUMBER = 256;
+  private final Injector frameworkInjector = Guice.createInjector(new TestKitFrameworkModule());
+
+  private final NativeHandle nativeHandle;
+  private final Map<Short, Service> services = new HashMap<>();
+
+  private TestKit(List<Class<? extends ServiceModule>> serviceModules, EmulatedNodeType nodeType,
+                  short validatorCount, @Nullable TimeProvider timeProvider) {
+    List<UserServiceAdapter> serviceAdapters = toUserServiceAdapters(serviceModules);
+    populateServiceMap(serviceAdapters);
+    boolean isAuditorNode = nodeType == EmulatedNodeType.AUDITOR;
+    UserServiceAdapter[] userServiceAdapters = serviceAdapters.toArray(new UserServiceAdapter[0]);
+    // TODO: fix after native implementation
+    nativeHandle = null;
+//    nativeHandle = new NativeHandle(
+//        nativeCreateTestKit(userServiceAdapters, isAuditorNode,
+//            validatorCount, timeProvider));
+  }
+
+  /**
+   * Creates a TestKit network with a single validator node for a single service.
+   */
+  public static TestKit forService(Class<? extends ServiceModule> serviceModule) {
+    return new TestKit(singletonList(serviceModule), EmulatedNodeType.VALIDATOR, (short) 0, null);
+  }
+
+  /**
+   * Returns an instance of a service with the given service id and service class.
+   *
+   * @return the service instance or null if there is no service with such id
+   * @throws NullPointerException if the service with given id was not found
+   * @throws IllegalArgumentException if the service could not be cast to given class
+   */
+  public <T extends Service> T getService(short serviceId, Class<T> serviceClass) {
+    Service service = services.get(serviceId);
+    checkNotNull(service, "Service with given id=%s was not found", serviceId);
+    checkArgument(service.getClass().equals(serviceClass),
+        "Service (id=%s, class=%s) cannot be cast to %s",
+        serviceId, service.getClass().getCanonicalName(), serviceClass.getCanonicalName());
+    return serviceClass.cast(service);
+  }
+
+  /**
+   * Creates a user service adapter from the service module.
```

Comment:
Instantiates a service given its module and wraps in a UserServiceAdapter
for the native code. 

(we don't create a USA from the module, but a service, and then wrap it).